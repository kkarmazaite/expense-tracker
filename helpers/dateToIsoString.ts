export const dateToIsoString = (date:Date) => {

  const pad = (number: number) => number < 10 ? '0' + number : number

  return `${date.getFullYear()}-${pad(date.getMonth() + 1)}-${pad(date.getDate())}`
}