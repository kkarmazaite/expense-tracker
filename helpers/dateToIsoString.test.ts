import { expect, describe, it } from 'vitest'
import { dateToIsoString } from './dateToIsoString'

describe('Helper | dateToIsoString', async() => {
  it('Check dateToIsoString function', () => {
    const input = new Date('2023-04-19')
    const output = '2023-04-19'

    expect(dateToIsoString(input)).toStrictEqual(output)
  })
})