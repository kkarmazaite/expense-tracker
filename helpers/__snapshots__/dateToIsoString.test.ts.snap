// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Helper | dateToIsoString > Check dateToIsoString function 1`] = `"2023-04-19"`;
